src/select/stateful-select-container.js

```diff
@@ -70,12 +80,11 @@ class StatefulSelectContainer extends React.Component<
     switch (type) {
       case STATE_CHANGE_TYPE.select:
       case STATE_CHANGE_TYPE.unselect:
-      case STATE_CHANGE_TYPE.clearAll:
         nextState = {
           selectedOptions: params.selectedOptions,
         };
         break;
-      case STATE_CHANGE_TYPE.keyDown: {
+      case STATE_CHANGE_TYPE.textInputChange: {
```

Comment:
Continuing the conversion from here: https://github.com/uber-web/baseui/pull/323#discussion_r221400458

It seems problematic to offer `textValue` as a prop in the stateless select if it's also stored in state internally. If a user tries to change the `textValue` prop after the component is constructed, it won't actually do anything. `textValue` should probably be removed from stateful component and only kept in `state` of stateless component. Can't think of a good use case for a user controlling this from outside.

`selectedOptions`, on the other hand, is the primary form control value and should probably live in stateful component and only be passed as a prop to stateless component (not stored in `state` there).

These changes might make more sense of a follow-up PR, but I think it's relevant to this PR because there's no point in renaming and offering `STATE_CHANGE_TYPE .textInputChange` if we're just going to remove it in a future PR.

Reply:
ok, changed

Reply:
Do we clear the text input's value when an option is selected? When the input's value is cleared in the case of multiselect?

Reply:
no, we do not. This request from Ahn seems to be controversial. I explained that commenting it in previous PR https://github.com/uber-web/baseui/pull/292#issuecomment-423381247

It can happen only in Search mode, however:
- we do not have design for single select mode. And I can't imagine how it should look like
- In multiple select (and may be in single) once we clear it, it will trigger close of DropDown (not sure if it's what expected) and will cause options to be reloaded as text query has changed